corrscope: 0.8.1 -> 0.9.0

OPNA2608 opened the pull request:
## Description of changes

Closes #295349

CC @FavoritoHJS

Only tested that it launches rn, can only do proper testing in afew hours.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SuperSandro2000 commented on pkgs/by-name/co/corrscope/package.nix:
```diff
@@ -1,49 +1,39 @@
   pname = "corrscope";
-  version = "0.8.1";
+  version = "0.9.0";
   format = "pyproject";
```
```suggestion
  pyproject = true;
```

OPNA2608 replied:
Applied.

SuperSandro2000 commented on pkgs/by-name/co/corrscope/package.nix:
```diff
@@ -1,49 +1,39 @@
 
   buildInputs = [
-    ffmpeg
+    ffmpeg-full
```
Shouldn't only ffmpeg be used and not full? What are we missing from the normal variant?

OPNA2608 replied:
https://github.com/NixOS/nixpkgs/issues/271863#issuecomment-1837484140

> `corrscope`: Uses `ffplay` for export preview, which depends on `withFullDeps`.

SuperSandro2000 replied:
see #296128

OPNA2608 replied:
Switched to regular `ffmpeg` now, and tested that it works fine.

OPNA2608 committed:
corrscope: 0.8.1 -> 0.9.0

OPNA2608 committed:
corrscope: Migrate to by-name

OPNA2608 commented:
@OfBorg eval

OPNA2608 committed:
corrscope: Switch to regular ffmpeg

ffplay is now enabled in the regular one. If a full ffmpeg is desired, then it can just be overridden.
